### DISCUSSION TITLE
Filtering sockets

### DISCUSSION THREAD

**Discussion by frusaka** [nortikin/sverchok#5002] [Q&A] [answered]
Not an issue. Just to ask, how do you filter the sockets that appear in the type menu in group nodes?
For more details, visit [This stack exchange question](https://blender.stackexchange.com/questions/301725/filtering-socket-in-custom-node-tree-inputs)

**Comment by Durman** [comment]
https://github.com/nortikin/sverchok/blob/d0d52173fa30692817b7043685e03c60f66acf1e/core/node_group.py#L266-L270

**Reply by frusaka** [reply]
Perfect! Thank you! But don't know why something like
`return isinstance(bpy.types.NodeSocket.bl_rna_get_subclass_py(socket_type),SvSocketCommon)`
does not work.

**Reply by Durman** [reply, marked as answer]
I can only guess. Blender does something with Python classes and they loose their hierarchy. So you can make isinstance test with exact class, not with its ancestor.
